jobs/jobscheduler.py

```diff
@@ -70,8 +71,23 @@ def runJob(job):
         job["exit_code"] = 0
     else:
         cmd = job["command"]
-        if ("gpu" in job) and (job["gpu"] >= 0):
-            cmd = cmd.replace("HIP_VISIBLE_DEVICES=0", "HIP_VISIBLE_DEVICES=%s" % job["gpu"])
+        if job.get("gpu", -1) >= 0:
+            maker, idx = job["gpu"].split(":", 1)
+            # Remove assignment of GPUs if it exists
+            cmd = re.sub(r"HIP_VISIBLE_DEVICES=\S+", "", cmd)
+            cmd = re.sub(r"CUDA_VISIBLE_DEVICES=\S+", "", cmd)
+
+            if maker == "cuda":
+                cmd = cmd.replace(
+                    "cmsDriver.py",
+                    "CUDA_VISIBLE_DEVICES=%s HIP_VISIBLE_DEVICES=0 cmsDriver.py" % idx,
```

Comment:
I'm not sure what this does, but keep in mind that `0` means "use the first GPU", not that there are 0 GPUs.
For that, pass an empty value.